danielMunteanu/src/main/java/danielCollectionTask/AbstractOrder.java

```diff
@@ -0,0 +1,86 @@
+package danielCollectionTask;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+public abstract class AbstractOrder implements Order {
+    private final String id;
+    private final LocalDateTime date;
+    private List<Item> items;
+
+    public AbstractOrder(List<Item> items) {
+        this.id = id();
+        this.date = LocalDateTime.now();
+        this.items = items;
+    }
+
+    private static String id() {
+        return String.valueOf(UUID.randomUUID());
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public LocalDateTime getDate() {
+        return date;
+    }
+
+    public List<Item> getItems() {
+        return items;
+    }
+
+    public void setItems(List<Item> items) {
+        this.items = items;
+    }
+
+    public int count(Item item) {
+        int nrofitems = 0;
+        for (Item uniqueItem : this.items) {
+            if (uniqueItem.getTitle().equals(item.getTitle())) {
+                nrofitems += 1;
+            }
+        }
+        return nrofitems;
+    }
+
+    public double sumByTitle(Item item) {
+        double totalSum = 0D;
+        for (Item uniqueItem : this.items) {
+            if (uniqueItem.getTitle().equals(item.getTitle())) {
+                totalSum += uniqueItem.getPrice();
+            }
+        }
+        return totalSum;
+    }
+
+    @Override
+    public double sum() {
+        double sum = 0D;
+        for (int i = 0; i < items.size(); i++) {
+            sum += items.get(i).getPrice();
+        }
+        return sum;
+    }
+
+    @Override
+    public double sum(List<Item> items) {
+        double sumaTotala = 0D;
```

Comment:
let's have a common patter for naming convention...totalSum sounds better

Reply:
done